src/audio/dai-zephyr.c

```diff
@@ -1359,37 +1355,38 @@ int dai_zephyr_multi_endpoint_copy(struct dai_data **dd, struct comp_dev *dev,
 		case -EPIPE:
 			/* DMA status can return -EPIPE and current status content if xrun occurs */
 			if (direction == SOF_IPC_STREAM_PLAYBACK)
-				comp_dbg(dev, "dai_zephyr_multi_endpoint_copy(): dma_get_status() underrun occurred, endpoint: %d ret = %u",
+				comp_dbg(dev, "dma_get_status() underrun occurred endpoint: %d ret = %u",
 					 i, ret);
 			else
-				comp_dbg(dev, "dai_zephyr_multi_endpoint_copy(): dma_get_status() overrun occurred, enpdoint: %d ret = %u",
+				comp_dbg(dev, "dma_get_status() overrun occurred, enpdoint: %d ret = %u",
 					 i, ret);
 			break;
 		default:
 			return ret;
 		}
 
-		avail_bytes = MIN(avail_bytes, stat.pending_length);
-		free_bytes = MIN(free_bytes, stat.free);
-	}
-
-	/* calculate minimum size to copy */
-	if (direction == SOF_IPC_STREAM_PLAYBACK) {
-		src_frames = audio_stream_get_avail_frames(&multi_endpoint_buffer->stream);
-		sink_frames = free_bytes / frame_bytes;
-	} else {
-		src_frames = avail_bytes / frame_bytes;
-		sink_frames = audio_stream_get_free_frames(&multi_endpoint_buffer->stream);
+		if (direction == SOF_IPC_STREAM_PLAYBACK) {
```

Comment:
Probably best to include an inline comment stating why we use frames aligned (since we are removing an existing comment).

Reply:
Fixed